webcomponents/tool/src/main/frontend/packages/sakai-rubrics/src/SakaiRubricCriterionEdit.js

```diff
@@ -23,130 +23,94 @@ export class SakaiRubricCriterionEdit extends RubricsElement {
+            <div class="modal-header">
+              <h1 class="modal-title fs-5" id="edit-criterion-${this.criterion.id}-label">${this.criterion.title}</h1>
+              <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+            </div>
+            <div class="modal-body">
+              <div class="mb-3">
+                <label class="label-rubrics" for="criterion-title-edit-${this.criterion.id}">
+                  ${this._i18n.criterion_title}
+                </label>
+                <input title="${this._i18n.criterion_title}" id="criterion-title-edit-${this.criterion.id}" type="text" value="${this.criterionClone.title}" maxlength="255">
+              </div>
+              <label class="label-rubrics" for="criterion-title-edit-${this.criterion.id}">
+                ${this.isCriterionGroup ? html`
+                  ${this._i18n.criterion_group_description}
+                ` : html`
+                  ${this._i18n.criterion_description}
+                `}
+              </label>
+              <sakai-editor id="criterion-title-edit-${this.criterion.id}"
+                toolbar="BasicText"
+                content="${ifDefined(this.criterionClone.description)}"
+                @changed=${this.updateCriterionDescription}
+                ?textarea=${this.textarea}>
+              </sakai-editor>
+            </div>
+            <div class="modal-footer">
+              <div>
+                <button class="btn btn-primary" type="button" @click=${this._saveEdit}>
+                  ${this._i18n.save}
+                </button>
+                <button class="btn btn-secondary" type="button" data-bs-dismiss="modal" @click=${this._cancelEdit}>
+                  ${this._i18n.cancel}
+                </button>
+              </div>
             </div>
           </div>
         </div>
       </div>
     `;
   }
 
-  cancelEdit(e) {
+  _saveEdit(e) {
 
     e.stopPropagation();
-    const trigger = document.getElementById(`edit-criterion-trigger-${this.dataset.criterionId}`);
-    bootstrap.Popover.getInstance(trigger).hide();
-    trigger.focus();
-  }
 
-  saveEdit(e) {
-
-    e.stopPropagation();
-
-    const title = e.target.closest(".popover-body").querySelector("input").value;
-    const description = e.target.closest("div.form").querySelector("sakai-editor").getContent();
+    const title = e.target.closest(".modal-content").querySelector("input").value;
+    const description = e.target.closest(".modal-content").querySelector("sakai-editor").getContent();
 
     const body = JSON.stringify([
       { "op": "replace", "path": "/title", "value": title },
       { "op": "replace", "path": "/description", "value": description },
     ]);
 
-    const url = `/api/sites/${this.dataset.siteId}/rubrics/${this.dataset.rubricId}/criteria/${this.dataset.criterionId}`;
+    const url = `/api/sites/${this.siteId}/rubrics/${this.rubricId}/criteria/${this.criterion.id}`;
```

Comment:
Do you think it would be better to read if the variable `const url` was more specific? Ex. `const CRITERION_UPDATE_URL` or what's more suitable here.

Reply:
Not in this case, no, because there's only one url used in the fetch here. If we had nested fetches going on I would agree, but in this case I think url is fine. I also don't want to start using upper case for const, like in Java. This is because we would end up with pretty much everything in upper case. I see const as like val in Scala. The default.